src/indieweb_utils/feeds/discovery.py

```diff
@@ -17,7 +17,9 @@ class FeedUrl:
     title: str
 
 
-def discover_web_page_feeds(url: str, user_mime_types: Optional[List[str]] = None) -> List[FeedUrl]:
+def discover_web_page_feeds(
+    url: str, user_mime_types: Optional[List[str]] = None, web_page_request: requests.Response = None
```

Comment:
If the goal is to use standard IO objects for parameters, I think it would make more sense to make `web_page_request` a string. This way it can be sourced from any library.